### PR TITLE
[ARM] Fix az resource list cannot use tag when there is a default location

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Fix `az resource list` cannot use tag when there is a default location
Issue: #9916 
Improve the error message of the conflict between tag and other filter conditions for `az resource list` command.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -504,9 +504,24 @@ def _deploy_arm_template_subscription_scope(cli_ctx,
                        deployment_name, properties, deployment_location)
 
 
-def _list_resources_odata_filter_builder(resource_group_name=None, resource_provider_namespace=None,
+def _list_resources_odata_filter_builder(cli=None, resource_group_name=None, resource_provider_namespace=None,
                                          resource_type=None, name=None, tag=None, location=None):
     """Build up OData filter string from parameters """
+    if tag:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py]: try to use `if tag is not None` to specify that there is parameter value for tag

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 2]
```diff
@@ -504,9 +504,24 @@ def _deploy_arm_template_subscription_scope(cli_ctx,
                        deployment_name, properties, deployment_location)
 
 
-def _list_resources_odata_filter_builder(resource_group_name=None, resource_provider_namespace=None,
+def _list_resources_odata_filter_builder(cli=None, resource_group_name=None, resource_provider_namespace=None,
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py]: Pls keep parameter consistent with original function. cli is not consistent.

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 3]
```diff
@@ -504,9 +504,24 @@ def _deploy_arm_template_subscription_scope(cli_ctx,
...
+        if cli and cli.data and isinstance(cli.data['safe_params'], list):
+            for arg_name in cli.data['safe_params']:
+                if arg_name in ['-l', '--l', '--location']:
+                    manual_input_location = True
+                    break
+        if manual_input_location is False:
+            location = None
+
+        if name or location:
+            raise IncorrectUsageError('you cannot use the tag filter with other filters')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py]: pls list `other filters` here

**Comment by Juliehzl** [comment]
Through reconsidering this scenario, we should keep consistent for the location argument. If they don't want to utilize the build-in type, we can define location by ourselves. But if we use, users should know the influence of such location type.

File: src/azure-cli/azure/cli/command_modules/resource/custom.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py]: Done

File: src/azure-cli/azure/cli/command_modules/resource/custom.py (hunk 2 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py]: Done

File: src/azure-cli/azure/cli/command_modules/resource/custom.py (hunk 3 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py]: Done

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 4]
```diff
@@ -507,6 +507,20 @@ def _deploy_arm_template_subscription_scope(cli_ctx,
 def _list_resources_odata_filter_builder(resource_group_name=None, resource_provider_namespace=None,
                                          resource_type=None, name=None, tag=None, location=None):
     """Build up OData filter string from parameters """
+    if tag is not None:
+        if resource_group_name:
+            raise IncorrectUsageError('you cannot use the tag filter with resource group filter'
+                                      '(If the default value for resource group is set, please clear it with \'az configure\' command first)')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py]: use "az configure" command

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/custom.py]: Done

**Comment by Juliehzl** [comment]
@zhoxing-ms Why do you close this PR?

**Pull request merged**